### PR TITLE
Add method to check if repo vulnerability alerts are enabled

### PR DESCRIPTION
**Pull request by lboynton**
Just reopening #2334, is anyone able to review what would need to be done integration test wise to get this merged?

### CODE REVIEW

**Review by nickfloyd — CHANGES_REQUESTED** [review]
Hey @lboynton thank you for your effort here and a huge apology for the delay in getting some 👀 on it.  I made a few comments about no longer needing the preview header for the API call (the preview was promoted in 09/2021) - so you can remove those and use the default if you'd like.

✅ As far as integration tests - I think you might be ok there - you've got coverage for the client there as well as unit tests... thanks for doing those! 

❇️ You could add the positive (204) and negative (404) tests for the GET endpoint
❇️ If you wanted to/ or had time you could add the other CRUP ops (PUT and DELETE)

❌  Also, there seems to be a merge conflict, but that should go away once you remove the preview changes.

Thanks again!  Looking forward to getting this in.

File: Octokit/Helpers/AcceptHeaders.cs [hunk 1]
```diff
@@ -54,6 +54,8 @@ public static class AcceptHeaders
 
         public const string VisibilityPreview = "application/vnd.github.nebula-preview+json";
 
+        public const string DependencyAlertsPreview = "application/vnd.github.dorian-preview+json";
```
[the remark this answers is not in the document]
Comment (lboy2) [reply Octokit/Helpers/AcceptHeaders.cs]: Removed

[the remark this answers is not in the document]
Comment (lboynton) [reply Octokit/Helpers/AcceptHeaders.cs]: Removed

File: Octokit/Clients/RepositoriesClient.cs [hunk 2]
```diff
@@ -208,6 +208,33 @@ public Task<Repository> Transfer(long repositoryId, RepositoryTransfer repositor
...
+        /// <summary>
+        /// Checks if vulnerability alerts are enabled for the specified repository.
+        /// </summary>
+        /// <remarks>
+        /// See the <a href="https://docs.github.com/en/rest/reference/repos#check-if-vulnerability-alerts-are-enabled-for-a-repository">API documentation</a> for more information.
+        /// </remarks>
+        /// <param name="owner">The current owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <returns>A <c>bool</c> indicating if alerts are turned on or not.</returns>
+        [Preview("dorian")]
```
[the remark this answers is not in the document]
Comment (lboynton) [reply Octokit/Clients/RepositoriesClient.cs]: Removed

File: Octokit.Tests/Clients/RepositoriesClientTests.cs [hunk 3]
```diff
@@ -357,6 +360,55 @@ public async Task SendsPreviewHeaderById()
...
+            [Theory]
+            [InlineData(HttpStatusCode.NoContent, true)]
+            [InlineData(HttpStatusCode.NotFound, false)]
+            public async Task RequestsCorrectValueForStatusCode(HttpStatusCode status, bool expected)
+            {
+                var response = CreateResponse(status);
+                var responseTask = Task.FromResult<IApiResponse<object>>(new ApiResponse<object>(response));
+                var connection = Substitute.For<IConnection>();
+                connection.Get<object>(Arg.Is<Uri>(u => u.ToString() == "repos/owner/name/vulnerability-alerts"),
+                    null, AcceptHeaders.DependencyAlertsPreview).Returns(responseTask);
```
[the remark this answers is not in the document]
Comment (lboynton) [reply Octokit.Tests/Clients/RepositoriesClientTests.cs]: Removed

File: Octokit/Clients/RepositoriesClient.cs [hunk 4]
```diff
@@ -208,6 +208,33 @@ public Task<Repository> Transfer(long repositoryId, RepositoryTransfer repositor
...
+        [ManualRoute("GET", "/repos/{owner}/{repo}/vulnerability-alerts")]
+        public async Task<bool> AreVulnerabilityAlertsEnabled(string owner, string name)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, nameof(owner));
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+
+            try
+            {
+                var response = await Connection.Get<object>(ApiUrls.RepositoryVulnerabilityAlerts(owner, name), null, AcceptHeaders.DependencyAlertsPreview).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (lboynton) [reply Octokit/Clients/RepositoriesClient.cs]: Removed

**Comment by lboynton** [comment]
Thanks for reviewing @nickfloyd! I've merged in the upstream changes in `main` to resolve the conflicts.

>You could add the positive (204) and negative (404) tests for the GET endpoint

I believe these are already there in the `TheAreVulnerabilityAlertsEnabledMethod` class?

>If you wanted to/ or had time you could add the other CRUP ops (PUT and DELETE)

I might leave that to a separate PR for me or someone else :)

**Review by nickfloyd — APPROVED** [review]
This looks good!  Thank you again for the contributions here! ✅

**Comment by lboynton** [comment]
Thanks! :)

**Comment by nickfloyd** [comment]
release_notes: Add method to check if repo vulnerability alerts are enabled

**Pull request merged**